### PR TITLE
Implements blockOffset top level query.

### LINKED ISSUE #69: Implement blockOffset top-level query
This query selects a block based on a hash or a number, and an offset. The offset can be positive, negative or zero.

*Validation:* either hash or number MUST be informed, but not both, and offset MUST be informed.

*Behaviour:* 
- If hash provided: call `getBlock` with the hash, add the offset to the block number that comes back, fetch that block and return it.
- If number provided: we just add the offset to the number and return that block.

*Tests:* 
- happy path: all permutations of positive, negative, zero offset; and hash, number.
- error scenarios: inexistent block, hash and number not informed, both hash and number informed.

### PR DESCRIPTION
**Pull request by akhila-raju**
Resolves #69.

### CODE REVIEW

**Comment by raulk** [comment]
@akhila-raju Overall LGTM.

But I wonder if there's room to refine the input parameters to make them less "pedantic" – this was my fault. I believe the schema has unnecessary redundancy, i.e. `offset: { number: 123, hash: "0x0123...", offset: 123 }`. 

One approach is to flatten out the parameters:

```
blockOffset(number: 1234, offset: -12) {
}
```

We can also take this opportunity to introduce a `tag: BlockTag` criterion like in other block queries.

WDYT?

**Comment by akhila-raju** [comment]
Yep - exactly what I was thinking. I'll make the fix.

File: test.js [hunk 1]
```diff
@@ -0,0 +1,8145 @@
+require = (function() {
```
Comment (raulk) [inline test.js]: What's this file?

Comment (akhila-raju) [reply test.js]: I'm not sure - it's from your commit

**Review by raulk — APPROVED** [review]
LGTM. Initial pull request was modified to allow for 0-valued offsets (to make life easier for queries using GraphQL variables to set a dynamic offset).

**Pull request merged**